tests/Query/DbTestCase.php

```diff
@@ -19,13 +25,24 @@ class DbTestCase extends TestCase
 
     public function setUp(): void
     {
-        $this->configuration = new Configuration();
-        $this->configuration->setMetadataCache(new ArrayAdapter());
-        $this->configuration->setQueryCache(new ArrayAdapter());
+        if (method_exists(ORMSetup::class, 'createAttributeMetadataConfiguration')) {
+            $this->configuration = ORMSetup::createAttributeMetadataConfiguration([__DIR__ . '/../Entities'], true);
+        } else {
+            $this->configuration = new Configuration();
+            $this->configuration->setMetadataDriverImpl(ORMSetup::createDefaultAnnotationDriver([__DIR__ . '/../../Entities']));
+        }
+
+        if (method_exists($this->configuration, 'setMetadataCacheImpl') && method_exists($this->configuration, 'setQueryCacheImpl')) {
+            $this->configuration->setMetadataCacheImpl(new ArrayAdapter());
+            $this->configuration->setQueryCacheImpl(new ArrayAdapter());
+        } else {
+            $this->configuration->setMetadataCache(new ArrayAdapter());
+            $this->configuration->setQueryCache(new ArrayAdapter());
+        }
```

Comment:
I don't understand what problem you're attempting to solve here.

Reply:
I forgot to check since when the setMetadataCache and setQueryCache methods exist.